integration_tests/src/main/python/array_test.py

```diff
@@ -89,19 +88,32 @@
         DoubleGen(special_cases=[], no_nans=True),
         string_gen, boolean_gen, date_gen, timestamp_gen]
 
-# Merged "test_nested_array_item" with this one since arrays as literals is supported
+
+byte_array_index_gen = ByteGen(min_val=-25, max_val=25, special_cases=[None])
+short_array_index_gen = ShortGen(min_val=-25, max_val=25, special_cases=[None])
+int_array_index_gen = IntegerGen(min_val=-25, max_val=25, special_cases=[None])
+long_array_index_gen = LongGen(min_val=-25, max_val=25, special_cases=[None])
```

Comment:
Is it worth adding a special case where we use a large long that will truncate to a valid index when truncated to an int?  This is to check for any regressions or change in behavior regarding handling of long indices.